openeo/rest/job.py

```diff
@@ -79,24 +80,61 @@ def list_results(self, type=None):
         # GET /jobs/{job_id}/results
         pass
 
-    def download_results(self, target: Union[str, pathlib.Path]) -> pathlib.Path:
-        """ Download job results."""
+    def download_results(self, target: Union[str, pathlib.Path] = None) -> list:
+        """
+        Download job results to the target folder path.
+        target is not set: it stores the result files at the execution path.
+        target is set to a file path including the filename: The result gets stored in this file. If there are more than
+                                                             one result file, only one gets downloaded into that filepath.
+        target is set to a folder path: All download files will be downloaded into this path.
+
+
+        :param target: String path, where to put the result files or filepath, if only one resultfile is expected.
+        :return: file_list: List of strings containing the downloaded file paths.
+        """
+
         results_url = "/jobs/{}/results".format(self.job_id)
         r = self.connection.get(results_url, expected_status=200)
-        links = r.json()["links"]
-        if len(links) != 1:
-            # TODO handle download of multiple files?
-            raise OpenEoClientException("Expected 1 result file to download, but got {c}".format(c=len(links)))
-        file_url = links[0]["href"]
-
-        target = pathlib.Path(target)
-        with target.open('wb') as handle:
-            response = self.connection.get(file_url, stream=True)
-            for block in response.iter_content(1024):
-                if not block:
-                    break
-                handle.write(block)
-        return target
+
+        file_list = []
+
```

Comment:
put a line  `target = Path(target or Path.cwd())` here, then you don't have to do `if target` anymore in the code below